ludwig/datasets/__init__.py

```diff
@@ -118,9 +126,87 @@ def load_dataset_uris(
             else:
                 test_set_out = _load_cacheable_dataset(test_set, backend)
 
+        return dataset_out, training_set_out, validation_set_out, test_set_out
+
+
+def _is_hf(dataset, training_set):
+    dataset_is_hf = dataset is not None and isinstance(dataset, str) and dataset.startswith(HF_PREFIX)
+    training_set_is_hf = (
+        training_set is not None and isinstance(training_set, str) and training_set.startswith(HF_PREFIX)
+    )
+    return dataset_is_hf or training_set_is_hf
+
+
+def _load_hf_datasets(
+    dataset: Optional[Union[str, DataFrame]],
+    training_set: Optional[Union[str, DataFrame]],
+    validation_set: Optional[Union[str, DataFrame]],
+    test_set: Optional[Union[str, DataFrame]],
+    backend: Backend,
+) -> Tuple[
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+]:
+    """Loads and returns any Hugging Face datasets as CacheableDataframes.
+
+    Returns the input unmodified for any non-HF datasets.
+    """
+    dataset_out = dataset
+    training_set_out = training_set
+    validation_set_out = validation_set
+    test_set_out = test_set
+
+    # Check that any of the datasets begin with the `hf://` prefix denoting a Hugging Face dataset URI
+    # Hugging Face datasets should follow the naming convention `hf://<hf_id>--<hf_subsample>`
+    if dataset is not None:
+        if isinstance(dataset, str) and dataset.startswith(HF_PREFIX):
+            dataset_out = _load_cacheable_hf_dataset(dataset, backend)
+        return dataset_out, training_set_out, validation_set_out, test_set_out
+
+    # Because of the conditional logic (_is_hf) in load_dataset_uris, if the above block is not triggered, then
+    # training_set must be a string that starts with HF_PREFIX
+    train_df = test_df = val_df = None
+    loader = get_dataset("hugging_face")
+    hf_id, hf_subsample = _get_hf_dataset_and_subsample(training_set)
+    train_df, val_df, test_df = loader.load(hf_id, hf_subsample, split=True)  # Call hugging_face loader
+    train_df = backend.df_engine.from_pandas(train_df)
+    training_set_out = CacheableDataframe(df=train_df, name=training_set, checksum=None)
+
+    if isinstance(validation_set, str) and validation_set.startswith(HF_PREFIX):
+        if validation_set == training_set:
+            # Reuse the loaded DF from the training split
+            val_df = backend.df_engine.from_pandas(val_df)
+            validation_set_out = CacheableDataframe(df=val_df, name=validation_set, checksum=None)
+        else:
+            validation_set_out = _load_cacheable_hf_dataset(validation_set, backend, split_set=VALIDATION)
```

Comment:
Add a commenthere that in general this handles an edge case where the user has specified different HF datasets for their dataset arguments. Not sure if this is a real use case, but potentially also worth adding a warning.